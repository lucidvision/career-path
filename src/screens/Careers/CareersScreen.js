import React, { Component } from 'react'
import { Keyboard, StyleSheet, TouchableWithoutFeedback, View } from 'react-native'
import { IconButton, SearchInput } from 'components'
import { SimpleLineIcons, Feather } from '@expo/vector-icons'
import { jobs } from 'data/jobs'

class Careers extends Component {
  handleCategoriesPressed = () => {
    this.props.navigation.navigate('CategoriesList')
  }
  handleFilterPressed = () => {
    this.props.navigation.navigate('Filter')
  }
  handleJobPressed = (job) => {
    this.props.navigation.navigate('Job', job)
  }
  render () {
    return (
      <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
        <View style={styles.container}>
          <SearchInput data={jobs} onItemPressed={this.handleJobPressed} />
          <IconButton onPress={this.handleCategoriesPressed} title='Categories'>
            <SimpleLineIcons name='list' size={25} />
          </IconButton>
          <IconButton onPress={this.handleFilterPressed} title='Filter'>
            <Feather name='filter' size={25} />
          </IconButton>
        </View>
      </TouchableWithoutFeedback>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-around',
    padding: 10
  }
})

export default Careers
