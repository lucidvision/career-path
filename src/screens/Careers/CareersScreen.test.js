import React from 'react'
import { shallow } from 'enzyme'
import CareersScreen from './CareersScreen'

const createCareersScreen = (override) => {
  const props = {...override}

  return shallow(<CareersScreen {...props} />)
}

describe('CareersScreen', () => {
  const navMock = jest.fn()
  const component = createCareersScreen({
    navigation: {
      navigate: navMock
    }
  })

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('handleCategoriesPressed', () => {
    it('navigates to the categories list screen', () => {
      component.find('IconButton').at(0).props().onPress()
      expect(navMock).toHaveBeenCalledWith('Categories')
    })
  })

  describe('handleFilterPressed', () => {
    it('navigates to the filter screen', () => {
      component.find('IconButton').at(1).props().onPress()
      expect(navMock).toHaveBeenCalledWith('Filter')
    })
  })

  describe('handleJobPressed', () => {
    it('navigates to the job screen', () => {
      component.find('SearchInput').props().onItemPressed({name: 'Doctor'})
      expect(navMock).toHaveBeenCalledWith('Job', {name: 'Doctor'})
    })
  })
})
