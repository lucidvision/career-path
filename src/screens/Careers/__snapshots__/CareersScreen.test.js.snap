// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareersScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 10,
    }
  }
>
  <SearchInput
    data={
      Array [
        Object {
          "educationRequired": "MD",
          "name": "Doctor",
          "salary": 100000,
          "score": 2,
        },
        Object {
          "educationRequired": "Bachelors",
          "name": "Nurse",
          "salary": 50000,
          "score": 1,
        },
        Object {
          "educationRequired": "Bachelors",
          "name": "Therapist",
          "salary": 50000,
          "score": 0,
        },
        Object {
          "educationRequired": "High School",
          "name": "Tour Guide",
          "salary": 30000,
          "score": 0,
        },
        Object {
          "educationRequired": "Bachelors",
          "name": "Agent",
          "salary": 50000,
          "score": 0,
        },
        Object {
          "educationRequired": "Bachelors",
          "name": "Hotel Manager",
          "salary": 80000,
          "score": 1,
        },
      ]
    }
    onItemPressed={[Function]}
  />
  <IconButton
    onPress={[Function]}
    title="Categories"
  >
    <Icon
      allowFontScaling={false}
      name="list"
      size={25}
    />
  </IconButton>
  <IconButton
    onPress={[Function]}
    title="Filter"
  >
    <Icon
      allowFontScaling={false}
      name="filter"
      size={25}
    />
  </IconButton>
</View>
`;
